app/pipeline.py

```
@@ -1 +1,138 @@
+                for item in self.scraped_data_objects
+            ]
+            
+            print(f"Scraping completed. Collected {len(self.scraped_data)} topics.")
+        except Exception as e:
+            print(f"Scraping failed: {e}")
+            raise
+
+    def run_translator(self):
+        print("\n=== Step 2: Translation ===")
+        if not hasattr(self, 'scraped_data'):
+            raise ValueError("Scraped data not found. Run scraper first.")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)